learningequality/studio
# Channel staging tree

## Description

Implement Vueified staging tree workflow. The only missing part is a toolbar that shouldn't contain any edit actions for staging tree. Currently, a toolbar for the root tree is used instead. I need to work on some general page layout updates first (that will affect more pages, so it should be done in a separate PR anyway) and then will be able to update the staging toolbar in a small follow-up PR. Taking into account that I am helping with Kolibri 0.14 and have the core logic ready for review, I've decided to open it now.

~I will be opening toolbar follow-up issues and one tech debt issue mentioned later in the evening.~
Toolbar follow-up issue #1985

#### Issue Addressed

https://www.notion.so/learningequality/Staged-channel-page-c1f66b973038492f8e2bce19781da9a8

#### Screenshots of root tree page updates

Ready to review banner:
![review-banner](https://user-images.githubusercontent.com/13509191/87544422-b8788b80-c6a6-11ea-951a-fecd30c36780.png)

Snackbar after a staging tree deployed:
![snackbar](https://user-images.githubusercontent.com/13509191/87544578-fe355400-c6a6-11ea-90a5-fad98c9624f3.png)

#### Screenshots of a new staging tree page

Resources list with preview drawer open:
![preview](https://user-images.githubusercontent.com/13509191/87545192-df838d00-c6a7-11ea-99d4-bd2a562c8426.png)

More resources (see the bottom bar and summary preview dialog):
![resources-added](https://user-images.githubusercontent.com/13509191/87545239-ee6a3f80-c6a7-11ea-9c17-8319ef07bea5.png)

Less resources (see the bottom bar and summary preview dialog):
![resources-removed](https://user-images.githubusercontent.com/13509191/87545296-004be280-c6a8-11ea-9eec-32e51b511f34.png)

No diff (see the bottom bar and summary preview dialog):
![no-diff](https://user-images.githubusercontent.com/13509191/87545323-0b9f0e00-c6a8-11ea-8e63-f6583d1d4a1e.png)

Deploy dialog:
![deploy-dialog](https://user-images.githubusercontent.com/13509191/87545354-15287600-c6a8-11ea-91d2-5648335977c5.png)

## Steps to Test

Visit demo server http://misrob-staging-channel.studio.cd.learningequality.org (user: "a@a.com", password: "a") where four test channels are prepared, two of them entirely new, and two of them that have been deployed already and updated later. You can try to deploy them (there will be two reviewers, so please leave some of the channels not deployed for the second reviewer if they'd like to check it too or ask me to run my test chef again). Alternatively, you can test the whole flow by running some chef against the demo server.

You can follow [Requirements](https://www.notion.so/learningequality/Staged-channel-page-c1f66b973038492f8e2bce19781da9a8#c9455e8852b74dc7ab3442be3f8852f4) when testing.

#### Does this introduce any tech-debt items?

Staging diff endpoint response optimization #1986

## Checklist
